python/lsst/obs/base/gen2to3/convertRepo.py

```diff
@@ -503,7 +503,10 @@ def run(self, root: str, collections: List[str], *,
         # to convert, because Gen2 alone doesn't know enough about the
         # relationships between data IDs.
         for converter in converters:
-            converter.insertDimensionData()
+            try:
+                converter.insertDimensionData()
```

Comment:
Could we have a `registry.insertDimensionData()` option for doing slow inserts where it calls queryDimensionData for each record first and complains if there is a difference? Then only inserting the ones that don't already exist?

Reply:
Yes, and I was already planning to add one on one of the "fix ingest" tickets.  We already have the underlying primitive in `Database.sync`, but we may need to think about how to identify which fields to use for lookup, which fields to compare, and which fields to just set when inserting.  We may already have all of the information we need for that in the dimension definitions, but I'm not certain.

The big catch with this is that each `sync` needs to start its own transaction, and hence can't be nested inside another transaction.  AFAICT, there is no generic-SQL way to implement it in a concurrent and transaction-nesting-friendly way.